### DISCUSSION TITLE
Is it possible to generate an iso from this flake?

### DISCUSSION THREAD

**Discussion by achyutbadyal** [richen604/hydenix#4] [Q&A] [answered]
Currently I am on arch linux and I want to switch to nix, my question is, is it possible to generate an ISO from this flake, to avoid the step of installing some other flavor first and then switch to this? I looked it up and found that it is possible to do so through ```nixos-generators``` (for which i need nix, which defeats the purpose of wanting to this). Is there any other way of achieving this? Sorry if this is a super newbie question, and thanks in advance.

**Comment by richen604** [comment, marked as answer]
@YuanziX You're spot on, nixos-generators is exactly what you would need. 
you would clone the project, add the nixos-generators flake as per docs. and generate the iso using [nixosConfigurations.hydenix](https://github.com/richen604/hydenix/blob/2abae07799d101f5eeed719c5a8c4d04f0c44635/flake.nix#L80)

you can do this all from arch using nix. i recommend you install from [nix-installer from DeterminateSystems](https://github.com/DeterminateSystems/nix-installer)

it would be a good idea to have iso's for this generate for releases, i plan on making a ci script for this eventually

**Reply by achyutbadyal** [reply]
I successfully generated the iso, booted into it using ventoy but it landed me on sddm and asked for password (I "correctly" guessed the password to be ```hydenix```) but when i enter the password it just turns black for a second and then reopens sddm. If I enter some other password then it shows ```incorrect password``` so hydenix must be the correct password but I couldn't figure out why i could not boot. Are you aware of any reason which might cause this?

**Reply by richen604** [reply]
@YuanziX what does journalctl say?

**Reply by richen604** [reply]
you can also tty and run sddm that way to see any output. then try Hyprland

**Reply by achyutbadyal** [reply]
Used tty and i was able to login correctly, but hyprland wouldn't launch.

\>>>>>> [very big logs here](https://bashupload.com/XNx6f/logs.log) <<<<<<

**Reply by achyutbadyal** [reply]
what I understand after looking at the logs is that sddm is working fine but hyprland is not launching.

**Reply by achyutbadyal** [reply]
The logs have some 'nouveau' stuff in them so I tried it on a system without a nvidia GPU in it and it booted fine, guess it's due to that then.

**Reply by richen604** [reply]
looking at the logs its opengl/hardware acceleration related. im guessing nvidia gpu? 

revise `hosts/nixos/configuration.nix` & setup nvidia (or other) drivers there. then regenerate iso

I'll create a special `drivers.nix` file with common setups and include it in the readme

nix doesnt really have a way to detect drivers ([besides hacks like nixGL](https://github.com/nix-community/nixGL/blob/310f8e49a149e4c9ea52f1adf70cdc768ec53f8a/nixGL.nix#L211))

**Reply by achyutbadyal** [reply]
I added the nvidia conf that I used from the time that I last used nix but it still didn't work, same issue. I guess I'll just install it normally, thanks for your help.
